File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int profit =0;
        int mn=INT_MAX;
        for(int i=0;i<n;i++){
            mn = min(mn, prices[i]);
            profit = max(profit, prices[i]-mn);
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        l=0
        r=1
        max_profit=0
        while(r<len(prices)):
            if(prices[l]<prices[r]):
                max_profit=max(max_profit, prices[r]-prices[l])
            else:
                l=r
            r+=1
            
        return max_profit
        